src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Operators.cs

```diff
@@ -427,9 +427,8 @@ private ExprBinOp bindNullEqualityComparison(ExpressionKind ek, BinOpArgInfo inf
                     return exprRes;
                 }
             }
-            Expr pExpr = BadOperatorTypesError(ek, info.arg1, info.arg2, GetTypes().GetErrorSym());
```

Comment:
I believe it should be possible to get rid of all use of `ErrorType` (which `GetErrorSym()` returns) outside of `MethodTypeInferrer` (from which it never escapes), which would rule out a lot of branches, including some which make for tricky reasoning about possible behaviour, but of course I have to **know** I can before I can hit the delete key.